eng/common/pipelines/templates/steps/docs-metadata-release.yml

```diff
@@ -6,33 +6,23 @@ parameters:
   RepoId: 'not-specified'
   WorkingDirectory: ''
   ScriptDirectory: eng/common/scripts
-  TargetDocRepo: ''
-  TargetDocBranch: 'smoke-test'
-  PRBranchName: 'not-specified'
+  TargetDocRepoName: ''
+  TargetDocRepoOwner: ''
+  PRBranchName: 'smoke-test-rdme'
   ArtifactName: ''
+  Language: ''
   ServiceDirectory: ''
+  DocRepoDestinationPath: '' #usually docs-ref-services/
 
 steps:
 - pwsh: |
-    git clone ${{ TargetDocRepository }} ${{ parameters.WorkingDirectory }}/repo
+    git clone https://github.com/${{ parameters.TargetDocRepoOwner }}/${{ parameters.TargetDocRepoName }} ${{ parameters.WorkingDirectory }}/repo
     
-    $sourceBranch = "${{ parameters.TargetDocBranch }}"
-  
     try {
-      Push-Location {{ parameters.WorkingDirectory }}/repo
+      Push-Location ${{ parameters.WorkingDirectory }}/repo
     
-      # 0 if exists, 1 if not
-      git show-ref --verify --quiet refs/heads/$sourceBranch
-      $sourceBranchExists = $LastExitCode
-
-      if (-not $sourceBranchExists) {
-        # there is an existing smoke-test branch we can base off of
-        Write-Host "git checkout $sourceBranch"
-        git checkout $sourceBranch
-      }
-
-      Write-Host "git checkout -b {{ parameters.PRBranchName }}"
-      git checkout -b ${{ parameters.PRBranchName }}
+      Write-Host "git checkout smoke-test"
+      git checkout smoke-test
```

Comment:
If there is ever a case where this branch doesn't exist in the repo this step will fail. Is that why you have the try/finally? Perhaps you have access to make this as a protected branch in the docs repo to help ensure it always exists.

Reply:
I intentionally designed it this way. `smoke-test` will always exist. I haven't taken the step to make it a protected branch as sometimes temporarily deletion is the easiest way to clean up the mountains of commited YML, but it will _always_ exist.